crates/syn2mas/src/migration.rs

```diff
@@ -177,6 +182,20 @@ async fn migrate_users(
         if bool::from(user.is_guest) {
             flags |= UserFlags::IS_GUEST;
         }
+        if user.appservice_id.is_some() {
+            flags |= UserFlags::IS_APPSERVICE;
+
+            // Special case for appservice users: we don't insert them into the database
+            // We just record the user's information in the state and continue
+            state.users.insert(
+                CompactString::new(&mas_user.username),
+                UserInfo {
+                    mas_user_id: Uuid::nil(),
```

Comment:
We *could* make this field an [`Option<NonNilUuid>`](https://docs.rs/uuid/latest/uuid/struct.NonNilUuid.html) (that would take the same space in memory) if we wanted to make sure that we never use a nil one?

Reply:
I like this idea, even if it does mean you will probably need some unwraps or so, it feels more 'direct' to explicitly unwrap than potentially let some accident happen and find out later, probably in a harder to debug way.
That said, the current approach is defensible and not a big deal but avoiding sentinel values has been generally a pretty productive idea for me, so I prefer it as a maintenance safety improvement.